java/src/main/java/com/powsybl/python/network/NetworkAreaDiagramUtil.java

```
@@ -30,32 +31,136 @@ public final class NetworkAreaDiagramUtil {
         try (StringWriter writer = new StringWriter()) {
-            writeSvg(network, voltageLevelIds, depth, writer);
+            writeSvg(network, voltageLevelIds, depth, writer, highNominalVoltageBound, lowNominalVoltageBound);
             return writer.toString();
         } catch (IOException e) {
             throw new UncheckedIOException(e);
         }
     }
 
-    static void writeSvg(Network network, List<String> voltageLevelIds, int depth, String svgFile) {
+    static void writeSvg(Network network, List<String> voltageLevelIds, int depth, String svgFile,
+                         Double highNominalVoltageBound, Double lowNominalVoltageBound) {
         try (Writer writer = Files.newBufferedWriter(Paths.get(svgFile), StandardCharsets.UTF_8)) {
-            writeSvg(network, voltageLevelIds, depth, writer);
+            writeSvg(network, voltageLevelIds, depth, writer, highNominalVoltageBound, lowNominalVoltageBound);
         } catch (IOException e) {
             throw new UncheckedIOException(e);
         }
     }
+
+    static VoltageLevelFilter getNominalVoltageFilter(Network network, List<String> voltageLevelIds,
+                                                      double highNominalVoltageBound,
+                                                      double lowNominalVoltageBound, int depth) {
+        Objects.requireNonNull(network);
+        Objects.requireNonNull(voltageLevelIds);
+        Set<VoltageLevel> startingSet = new HashSet<>();
+
+        for (String voltageLevelId : voltageLevelIds) {
+            VoltageLevel vl = network.getVoltageLevel(voltageLevelId);
+            if (vl == null) {
+                throw new PowsyblException("Unknown voltage level id '" + voltageLevelId + "'");
+            }
+            if (highNominalVoltageBound > 0) {
+                if (vl.getNominalV() < lowNominalVoltageBound || vl.getNominalV() > highNominalVoltageBound) {
+                    throw new PowsyblException("vl '" + voltageLevelId +
```

Do we really want to throw an exception here? Couldn't a log warn be enough?

---

well not sure if we select a initial voltage level out of the voltage bounds it won't work anyway so I think if we don't throw an error there will be an empty diagram

---

I agree: if all your input voltage levels are out of bounds, you will end up with an empty diagram (and some warnings). But it may be harsch for users to get an exception for giving only one voltage level out of bounds.